tmt/steps/report/reportportal.py

```diff
@@ -91,77 +139,222 @@ class ReportReportPortal(tmt.steps.report.ReportPlugin[ReportReportPortalData]):
+        self.debug("Response code from the endpoint", str(response.status_code))
+        self.debug("Message from the endpoint", str(response.text))
 
     def go(self) -> None:
         """
-        Report test results to the server
+        Report test results to the endpoint
 
         Create a ReportPortal launch and its test items,
         fill it with all parts needed and report the logs.
         """
 
+        # TODO: ? why it did not closed :
+        #           tmt run discover provision -h connect -g 10.0.185.29 -u root
+        #               -p fo0m4nchU  prepare report -h reportportal --suite-per-plan
+        #               --defect-type idle -vvv finish tests -n .
+        #       * test uploading to idle tests
+        #       * resolve the problem with mypy
+        #       * replace rp_uuid with an universal structure for tmt plugin data
+        #       * check the problem with --upload-to-suite functonality
+        #       * check the param per plan, and do the matching when uploading (launch_uuid)
+        #       * check the defect_type, idle functionality
+        #       * upload documentation (help and spec)
+        #       * add the tests for new features
+        #       * add uploading files
+        #       * edit schemas
+
+        #       * check optionality in arguments
+        #       * restrict combinations + set warning
+        #                   - forbid rerun && launch-per-plan
+        #                   - forbid upload-to-suite && launch-per-plan
+        #       * try read a launch_uuid at first plan (self.step.plan.report.launch_uuid)
+
         super().go()
 
-        server = self.data.url
-        if not server:
-            raise tmt.utils.ReportError("No ReportPortal server url provided.")
-        server = server.rstrip("/")
+        endpoint = self.get("url")
+        if not endpoint:
+            raise tmt.utils.ReportError("No ReportPortal endpoint url provided.")
+        endpoint = endpoint.rstrip("/")
 
-        project = self.data.project
+        project = self.get("project")
         if not project:
             raise tmt.utils.ReportError("No ReportPortal project provided.")
 
-        token = self.data.token
+        token = self.get("token")
         if not token:
             raise tmt.utils.ReportError("No ReportPortal token provided.")
 
-        assert self.step.plan.name is not None
-        launch_name = self.data.launch or self.step.plan.name
-
-        url = f"{server}/api/{self.DEFAULT_API_VERSION}/{project}"
+        url = f"{endpoint}/api/{self.DEFAULT_API_VERSION}/{project}"
         headers = {
             "Authorization": "bearer " + token,
             "accept": "*/*",
             "Content-Type": "application/json"}
 
-        envar_pattern = self.data.exclude_variables
+        launch_time = str(int(time() * 1000))
+
+        # to support idle tests
+        executed = False
+        if len(self.step.plan.execute.results()) > 0:
```

Comment:
`if self.step.plan.execute.results():` is more Pythonic - empty list would be false-ish.

```python
executed = bool(self.step.plan.execute.results())
if executed:
    launch_time = self.step.plan.execute.results()[0].start_time
```

Reply:
@4N0body5 is this resolved?